service/storage_proxy.cc

```diff
@@ -152,7 +152,7 @@ static future<rpc::tuple<Elements..., replica::exception_variant>> encode_replic
 // Without it only local queries are available.
 class storage_proxy::remote {
     storage_proxy& _sp;
-    shared_ptr<migration_manager> _mm;
+    migration_manager* _mm;
```

Comment:
I don't have a strong opinion but I'd side with safety here, in case due to an initialization order bug migration manager is destroyed out of order. Using a shared pointer increases the chances that we get a nice exception not a crash. uninit() can clear the pointer just like in the current patch, and all other parts of the patch seem fine. Please feel free to use your own judgement here though.